lib/block_view.go

```diff
@@ -3920,6 +3920,89 @@ func (bav *UtxoView) _connectTransaction(
 	return utxoOpsForTxn, totalInput, totalOutput, fees, nil
 }
 
+func (bav *UtxoView) ConnectTransactions(
```

Comment:
I wish there was a comment here explaining why we need this. You obviously added it for a reason and it would be super helpful to have that reason here so that when I review I can internalize what drove the addition quickly instead of having to figure it out. poolcoke has recently internalized this advice and started writing extremely long comments on everything. I love this and would like to see more of it from everyone. See here as an example all throughout this file:
* https://github.com/deso-protocol/core/blob/3dc58f301235dfc663577c8182b1f85aa0bf0f95/lib/block_view_lockups.go#L114